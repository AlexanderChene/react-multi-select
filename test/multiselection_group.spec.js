import React from "react";
import { shallow, mount } from "enzyme";

import ListGroup from "../../../src/components/multiselectionlist/multiselection_group";
import ShallowRenderer from "react-test-renderer/shallow";

const renderer = new ShallowRenderer();

describe("ListGroup component", () => {
  test("renders by default", () => {
    const component = renderer.render(<ListGroup/>);
    expect(component).toMatchSnapshot();
  });

  test("renders by input props", () => {
    const inputProps = {
      label: "testLabel",
      groupLink: "testSelectGroupLabel"
    };
    const component = renderer.render(<ListGroup {...inputProps} />);
    expect(component).toMatchSnapshot();
  });

  test("handles onGroupLinkClick right", () => {
    let isOnGroupLinkClick = false;
    const onGroupLinkClick = () => (isOnGroupLinkClick = true);
    const listGroup = mount(<ListGroup onGroupLinkClick={onGroupLinkClick} />);

    listGroup
      .find("a")
      .props()
      .onClick();
    expect(isOnGroupLinkClick).toBe(true);
  });
});
