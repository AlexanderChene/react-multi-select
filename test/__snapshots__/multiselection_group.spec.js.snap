// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListGroup component renders by default 1`] = `
<li
  className="kn-multiselection_group__list_group"
>
  <a
    className="kn-multiselection_group__group_select_all"
    onClick={[Function]}
  >
    Select All
  </a>
  <BootstrapTooltip
    forceDelay={300}
    forcePlacement="bottom"
    tooltipContent="Group"
  >
    <span>
      Group
    </span>
  </BootstrapTooltip>
</li>
`;

exports[`ListGroup component renders by input props 1`] = `
<li
  className="kn-multiselection_group__list_group"
>
  <a
    className="kn-multiselection_group__group_select_all"
    onClick={[Function]}
  >
    testSelectGroupLabel
  </a>
  <BootstrapTooltip
    forceDelay={300}
    forcePlacement="bottom"
    tooltipContent="testLabel"
  >
    <span>
      testLabel
    </span>
  </BootstrapTooltip>
</li>
`;
